src/grpc/grpc_utils.h

```diff
@@ -38,6 +38,25 @@
 
 namespace triton { namespace server { namespace grpc {
 
+// The step of processing that the state is in. Every state must
+// recognize START, COMPLETE and FINISH and the others are optional.
+typedef enum {
+  START,
+  COMPLETE,
+  FINISH,
+  ISSUED,
+  READ,
+  WRITEREADY,
+  WRITTEN,
+  WAITING_NOTIFICATION,
+  CANCELLATION_ISSUED,
+  CANCELLED,
+  PARTIAL_COMPLETION
+} Steps;
```

Comment:
Can you add short but descriptive comments to the states here? Especially the new ones. For example, explaining the purpose of PARTIAL_COMPLETION, and when/why it's used. This will make it more maintainable for others in the future.